napari/_qt/containers/_layer_delegate.py

```diff
@@ -189,9 +195,66 @@ def editorEvent(
                 return model.setData(
                     index, state, Qt.ItemDataRole.CheckStateRole
                 )
+        # catch alt-click on the vis checkbox and hide *other* layer visibility
+        # on second alt-click, restore the visibility state of the layers
+        if event.type() == QMouseEvent.MouseButtonRelease and (
+            event.button() == Qt.MouseButton.LeftButton
+            and event.modifiers() == Qt.AltModifier
+        ):
+            self.initStyleOption(option, index)
+            style = option.widget.style()
+            check_rect = style.subElementRect(
+                style.SubElement.SE_ItemViewItemCheckIndicator,
+                option,
+                option.widget,
+            )
+            if check_rect.contains(event.pos()):
+                return self._show_on_alt_click_hide_others(model, index)
+
         # refer all other events to the QStyledItemDelegate
         return super().editorEvent(event, model, option, index)
 
+    def _show_on_alt_click_hide_others(
+        self,
+        model: QtCore.QAbstractItemModel,
+        index: QtCore.QModelIndex,
+    ) -> QtCore.QAbstractItemModel:
+        """On alt/option click of a layer show the layer, hide other layers,
+        and preserve current visibility of layer list, so it can be restored
+        once a layer is alt/option-clicked a second time.
+        """
+        clicked_layer = index.data(ItemRole)
+        layer_list: LayerList = model.sourceModel()._root
+        # show the alt-clicked layer
+        state = Qt.CheckState.Checked
+        if not self._layer_visibility_states:
+            # first option-click, so store visibility, hide others
+            for layer in layer_list:
+                self._layer_visibility_states[layer] = layer.visible
+                layer.visible = layer == clicked_layer  # hide others
+            # make a note that this layer was alt-clicked
+            self._alt_click_layer = ref(clicked_layer)
+        elif self._alt_click_layer() == clicked_layer:
```

Comment:
hm. maybe we should use `is` instead of `==`. As far as Know it will be same result in this case but will be more correct in the context of semantics.